### PR TITLE
Fix generating bindings for multibindings with class keys.

### PR DESCRIPTION
**Pull request by sidius**
Fixes https://github.com/JakeWharton/dagger-reflect/issues/132

Problem: wrong key for multibindings with class keys. The key has a value of `java.util.Map<java.lang.Class, com.example.I>` when it should be `java.util.Map<java.lang.Class<?>, com.example.I>`.

Solution: fix `ReflectiveModuleParser`: use generic return type when constructing `entryKeyType`.

### CODE REVIEW

File: reflect/src/main/java/dagger/reflect/ReflectiveModuleParser.java [hunk 1]
```diff
@@ -1,13 +1,6 @@
...
 
-import static dagger.reflect.Reflection.boxIfNecessary;
-import static dagger.reflect.Reflection.findAnnotation;
-import static dagger.reflect.Reflection.findMapKey;
-import static dagger.reflect.Reflection.findQualifier;
-import static dagger.reflect.Reflection.findScope;
-import static dagger.reflect.Reflection.findScopes;
-import static dagger.reflect.Reflection.maybeInstantiate;
-import static dagger.reflect.Reflection.requireAnnotation;
+import static dagger.reflect.Reflection.*;
```
Comment (runningcode) [inline reflect/src/main/java/dagger/reflect/ReflectiveModuleParser.java]: wildcard imports are not allowed in google java format. i'm surprised this passed verification.

Comment (sidius) [reply reflect/src/main/java/dagger/reflect/ReflectiveModuleParser.java]: Yeah, weird that `./gradlew googleJavaFormat` was fine with it. Reverted the change.

**Comment by JakeWharton** [comment]
Thanks!

**Comment by JakeWharton** [comment]
Is this blocking you from using the library? If so I can release a 0.1.1. Otherwise I'll sit on it for a bit until a 0.2.0.

**Comment by sidius** [comment]
> Is this blocking you from using the library? If so I can release a 0.1.1. Otherwise I'll sit on it for a bit until a 0.2.0.

I am using the snapshot version so there is no need to release now.

**Pull request merged**